data/src/main/java/org/apache/iceberg/data/BaseDeleteLoader.java

```diff
@@ -151,6 +151,7 @@ public PositionDeleteIndex loadPositionDeletes(
     return PositionDeleteIndexUtil.merge(deletes);
   }
 
+  @SuppressWarnings("CollectionUndefinedEquality")
```

Comment:
`CharSequenceMap` uses `CharSequenceWrapper` for consistent
 hashing and equals behaviour . The tool cannot figure out that.